### ISSUE TITLE
[BUG]  Missing and incomplete Metadata

### ISSUE THREAD

**Issue by chippey** [OpenImageIO/oiio#2864]
**Describe the bug**
When reading metadata from a file (in my case, a TIFF file), some metadata is missing (most GPS related info), and some is incomplete (some metadata that is Matrix, or 2D/3D vectors/points/tuples), only reading the first element.

**To Reproduce**
1. Run `iinfo -a -v` on the image file
2. Compare output to exiftool

**Expected behavior**
Full multi-element metadata values to be present, and output the missing metadata such as GPS.

**Evidence**
In particular, note the difference in output between iinfo and exiftool for the "AlignMatrix", "AlignXYDeg", "VignettingCenter", and "VignettingPolynomial" fields.  The "PerspectiveDistortion" key does seem to be correctly read in both iinfo and exiftool.

Output of exiftool (note, there is some filesystem related info that exiftool slaps on the output that I wouldn't expect OIIO to recreate.  Also, only one subimage's metadata is pasted for brevity):
```
ExifTool Version Number         : 12.14
File Name                       : IMG_0157.tif
Directory                       : /mnt/storage/EnSpec/DATA_Projects/PanelLocTestData/2020-07-19_17-23-11/multiband
File Size                       : 23 MiB
File Modification Date/Time     : 2018:08:01 01:55:22-05:00
File Access Date/Time           : 2021:02:14 17:33:46-06:00
File Inode Change Date/Time     : 2021:02:12 16:48:16-06:00
File Permissions                : rwxr--r--
File Type                       : TIFF
File Type Extension             : tif
MIME Type                       : image/tiff
Exif Byte Order                 : Little-endian (Intel, II)
Image Width                     : 1904
Image Height                    : 1428
Bits Per Sample                 : 12
Compression                     : Uncompressed
Photometric Interpretation      : BlackIsZero
Orientation                     : Horizontal (normal)
Samples Per Pixel               : 1
Rows Per Strip                  : 1428
X Resolution                    : 238
Y Resolution                    : 238
Planar Configuration            : Chunky
Resolution Unit                 : inches
Above Ground Altitude           : 111.612001
Flight Unique ID                : 2020-07-19_17-23-11
GPSXY Accuracy                  : 0.660000
GPSZ Accuracy                   : 1.017000
Image Unique ID                 : 5010
Is Normalized                   : 0
Model Type                      : perspective
Perspective Distortion          : -0.117,0.179,-0.045,0.000,0.000
Perspective Focal Length        : 7.800
Perspective Focal Length Units  : mm
Pitch                           : 0.000000
Principal Point                 : 3.263,2.471
Rig Camera Index                : 0
Rig Name                        : Sentera-6x
Rig Relatives                   : 0.000,0.000,0.000
Roll                            : 0.000000
Yaw                             : 90.000000
Band Name                       : Blue
Band Sensitivity                : 0.555
Central Wavelength              : 475
Sun Sensor                      : 16290.876
Wavelength FWHM                 : 30
Align Enabled                   : 1
Session                         : 2020-07-19_17-23-11
Vignetting Enabled              : 1
Align Matrix                    : 0.999999, 0.001538, 77.199890, -0.001538, 0.999999, 53.255005, 0.000000, 0.000000, 1.000000
Align XY Deg                    : 78.300, 51.800, -0.088
Vignetting Center               : 1024.00, 768.00
Vignetting Polynomial           : 0, 0, 6.5e-10
Fill Order                      : Normal
Make                            : Sentera
Camera Model Name               : 21214-01_3.2MP-GS-0001
Strip Offsets                   : 18972672
Strip Byte Counts               : 4078368
Software                        : 1.3.2
Modify Date                     : 2020:07:19 17:36:28
Exposure Time                   : 1/830
F Number                        : 1.8
ISO                             : 193
Exif Version                    : 0230
Date/Time Original              : 2020:07:19 17:36:28
Create Date                     : 2020:07:19 17:36:28
Aperture Value                  : 1.8
Metering Mode                   : Average
Focal Length                    : 8.0 mm
Image Number                    : 157
Sub Sec Time                    : 125538
Sub Sec Time Original           : 125538
Sub Sec Time Digitized          : 125538
Flashpix Version                : 0100
Focal Plane X Resolution        : 2898.5504
Focal Plane Y Resolution        : 2898.5504
Focal Plane Resolution Unit     : cm
Exposure Mode                   : Auto
Scene Capture Type              : Standard
GPS Version ID                  : 2.2.0.0
GPS Latitude Ref                : North
GPS Longitude Ref               : West
GPS Altitude Ref                : Above Sea Level
GPS Time Stamp                  : 17:36:28.125438
GPS Satellites                  : 26
GPS Measure Mode                : 3-Dimensional Measurement
GPS Dilution Of Precision       : 1.05
GPS Map Datum                   : EGM96
GPS Processing Method           : 3D
GPS Date Stamp                  : 2020:07:19
GPS Differential                : No Correction
GPS Horizontal Positioning Error: 0.66 m
Serial Number                   : 047
Lens Model                      : 8.00mm-0005_0024
Aperture                        : 1.8
Image Size                      : 1904x1428
Lens ID                         : 8.00mm-0005_0024
Megapixels                      : 2.7
Scale Factor To 35 mm Equivalent: 5.3
Shutter Speed                   : 1/830
Create Date                     : 2020:07:19 17:36:28.125538
Date/Time Original              : 2020:07:19 17:36:28.125538
Modify Date                     : 2020:07:19 17:36:28.125538
GPS Altitude                    : 473.7 m Above Sea Level
GPS Date/Time                   : 2020:07:19 17:36:28.125438Z
GPS Latitude                    : 42 deg 53' 41.20" N
GPS Longitude                   : 89 deg 43' 7.64" W
Circle Of Confusion             : 0.006 mm
Field Of View                   : 46.2 deg
Focal Length                    : 8.0 mm (35 mm equivalent: 42.2 mm)
GPS Position                    : 42 deg 53' 41.20" N, 89 deg 43' 7.64" W
Hyperfocal Distance             : 6.24 m
Light Value                     : 10.4

```

Output of iinfo:
```/mnt/storage/EnSpec/DATA_Projects/PanelLocTestData/2020-07-19_17-23-11/multiband/IMG_0157.tif : 1904 x 1428, 1 channel, uint12 tiff
    5 subimages: 1904x1428 [u12], 1904x1428 [u12], 1904x1428 [u12], 1904x1428 [u12], 1904x1428 [u12]
 subimage  0: 1904 x 1428, 1 channel, uint12 tiff
    channel list: Y
    compression: "none"
    DateTime: "2020:07:19 17:36:28"
    ExposureTime: 0.001409
    FNumber: 1.8
    Make: "Sentera"
    Model: "21214-01_3.2MP-GS-0001"
    Orientation: 1 (normal)
    PixelAspectRatio: 1
    planarconfig: "contig"
    ResolutionUnit: "in"
    Software: "1.3.2"
    XResolution: 238
    YResolution: 238
    Camera:AboveGroundAltitude: "111.612001"
    Camera:BandName: "Blue"
    Camera:BandSensitivity: "0.555"
    Camera:CentralWavelength: "475"
    Camera:FlightUniqueID: "2020-07-19_17-23-11"
    Camera:GPSXYAccuracy: "0.660000"
    Camera:GPSZAccuracy: "1.017000"
    Camera:ImageUniqueID: "5010"
    Camera:IsNormalized: "0"
    Camera:ModelType: "perspective"
    Camera:PerspectiveDistortion: "-0.117,0.179,-0.045,0.000,0.000"
    Camera:PerspectiveFocalLength: "7.800"
    Camera:PerspectiveFocalLengthUnits: "mm"
    Camera:Pitch: "0.000000"
    Camera:PrincipalPoint: "3.263,2.471"
    Camera:RigCameraIndex: "0"
    Camera:RigName: "Sentera-6x"
    Camera:RigRelatives: "0.000,0.000,0.000"
    Camera:Roll: "0.000000"
    Camera:SunSensor: "16290.876"
    Camera:WavelengthFWHM: "30"
    Camera:Yaw: "90.000000"
    Exif:ApertureValue: 1.69599 (f/1.8)
    Exif:BodySerialNumber: "047"
    Exif:DateTimeDigitized: "2020:07:19 17:36:28"
    Exif:DateTimeOriginal: "2020:07:19 17:36:28"
    Exif:ExposureMode: 0 (auto)
    Exif:FocalLength: 8 (8 mm)
    Exif:FocalPlaneResolutionUnit: 3 (cm)
    Exif:FocalPlaneXResolution: 2898.55
    Exif:FocalPlaneYResolution: 2898.55
    Exif:LensModel: "8.00mm-0005_0020"
    Exif:MeteringMode: 1 (average)
    Exif:SceneCaptureType: 0 (standard)
    Exif:SubsecTime: "125436"
    Exif:SubsecTimeDigitized: "125436"
    Exif:SubsecTimeOriginal: "125436"
    oiio:BitsPerSample: 12
    oiio:ColorSpace: "sRGB"
    Sentera:AlignEnabled: "1"
    Sentera:AlignMatrix: "1.000000"
    Sentera:AlignXYDeg: "-0.088"
    Sentera:Session: "2020-07-19_17-23-11"
    Sentera:VignettingCenter: "768.00"
    Sentera:VignettingEnabled: "1"
    Sentera:VignettingPolynomial: "6.5e-10"
    tiff:Compression: 1
    tiff:PhotometricInterpretation: 1
    tiff:PlanarConfiguration: 1
    tiff:RowsPerStrip: 2
 subimage  1: 1904 x 1428, 1 channel, uint12 tiff
    channel list: Y
    compression: "none"
    DateTime: "2020:07:19 17:36:28"
    ExposureTime: 0.001484
    FNumber: 1.8
    Make: "Sentera"
    Model: "21214-01_3.2MP-GS-0001"
    Orientation: 1 (normal)
    PixelAspectRatio: 1
    planarconfig: "contig"
    ResolutionUnit: "in"
    Software: "1.3.2"
    XResolution: 238
    YResolution: 238
    Camera:AboveGroundAltitude: "111.612001"
    Camera:BandName: "Green"
    Camera:BandSensitivity: "0.598"
    Camera:CentralWavelength: "550"
    Camera:FlightUniqueID: "2020-07-19_17-23-11"
    Camera:GPSXYAccuracy: "0.660000"
    Camera:GPSZAccuracy: "1.017000"
    Camera:ImageUniqueID: "5010"
    Camera:IsNormalized: "0"
    Camera:ModelType: "perspective"
    Camera:PerspectiveDistortion: "-0.117,0.179,-0.045,0.000,0.000"
    Camera:PerspectiveFocalLength: "7.800"
    Camera:PerspectiveFocalLengthUnits: "mm"
    Camera:Pitch: "0.000000"
    Camera:PrincipalPoint: "3.375,2.459"
    Camera:RigCameraIndex: "1"
    Camera:RigName: "Sentera-6x"
    Camera:RigRelatives: "0.000,0.000,0.000"
    Camera:Roll: "0.000000"
    Camera:SunSensor: "22948.891"
    Camera:WavelengthFWHM: "20"
    Camera:Yaw: "90.000000"
    Exif:ApertureValue: 1.69599 (f/1.8)
    Exif:BodySerialNumber: "047"
    Exif:DateTimeDigitized: "2020:07:19 17:36:28"
    Exif:DateTimeOriginal: "2020:07:19 17:36:28"
    Exif:ExposureMode: 0 (auto)
    Exif:FocalLength: 8 (8 mm)
    Exif:FocalPlaneResolutionUnit: 3 (cm)
    Exif:FocalPlaneXResolution: 2898.55
    Exif:FocalPlaneYResolution: 2898.55
    Exif:LensModel: "8.00mm-0005_0021"
    Exif:MeteringMode: 1 (average)
    Exif:SceneCaptureType: 0 (standard)
    Exif:SubsecTime: "125399"
    Exif:SubsecTimeDigitized: "125399"
    Exif:SubsecTimeOriginal: "125399"
    oiio:BitsPerSample: 12
    oiio:ColorSpace: "sRGB"
    Sentera:AlignEnabled: "1"
    Sentera:AlignMatrix: "1.000000"
    Sentera:AlignXYDeg: "0.303"
    Sentera:Session: "2020-07-19_17-23-11"
    Sentera:VignettingCenter: "768.00"
    Sentera:VignettingEnabled: "1"
    Sentera:VignettingPolynomial: "6.5e-10"
    tiff:Compression: 1
    tiff:PhotometricInterpretation: 1
    tiff:PlanarConfiguration: 1
    tiff:RowsPerStrip: 2
 subimage  2: 1904 x 1428, 1 channel, uint12 tiff
    channel list: Y
    compression: "none"
    DateTime: "2020:07:19 17:36:28"
    ExposureTime: 0.000902
    FNumber: 1.8
    Make: "Sentera"
    Model: "21214-01_3.2MP-GS-0001"
    Orientation: 1 (normal)
    PixelAspectRatio: 1
    planarconfig: "contig"
    ResolutionUnit: "in"
    Software: "1.3.2"
    XResolution: 238
    YResolution: 238
    Camera:AboveGroundAltitude: "111.612001"
    Camera:BandName: "Red"
    Camera:BandSensitivity: "0.693"
    Camera:CentralWavelength: "670"
    Camera:FlightUniqueID: "2020-07-19_17-23-11"
    Camera:GPSXYAccuracy: "0.660000"
    Camera:GPSZAccuracy: "1.017000"
    Camera:ImageUniqueID: "5010"
    Camera:IsNormalized: "0"
    Camera:ModelType: "perspective"
    Camera:PerspectiveDistortion: "-0.117,0.179,-0.045,0.000,0.000"
    Camera:PerspectiveFocalLength: "7.800"
    Camera:PerspectiveFocalLengthUnits: "mm"
    Camera:Pitch: "0.000000"
    Camera:PrincipalPoint: "3.245,2.443"
    Camera:RigCameraIndex: "2"
    Camera:RigName: "Sentera-6x"
    Camera:RigRelatives: "0.000,0.000,0.000"
    Camera:Roll: "0.000000"
    Camera:SunSensor: "14802.615"
    Camera:WavelengthFWHM: "30"
    Camera:Yaw: "90.000000"
    Exif:ApertureValue: 1.69599 (f/1.8)
    Exif:BodySerialNumber: "047"
    Exif:DateTimeDigitized: "2020:07:19 17:36:28"
    Exif:DateTimeOriginal: "2020:07:19 17:36:28"
    Exif:ExposureMode: 0 (auto)
    Exif:FocalLength: 8 (8 mm)
    Exif:FocalPlaneResolutionUnit: 3 (cm)
    Exif:FocalPlaneXResolution: 2898.55
    Exif:FocalPlaneYResolution: 2898.55
    Exif:LensModel: "8.00mm-0005_0022"
    Exif:MeteringMode: 1 (average)
    Exif:SceneCaptureType: 0 (standard)
    Exif:SubsecTime: "125690"
    Exif:SubsecTimeDigitized: "125690"
    Exif:SubsecTimeOriginal: "125690"
    oiio:BitsPerSample: 12
    oiio:ColorSpace: "sRGB"
    Sentera:AlignEnabled: "1"
    Sentera:AlignMatrix: "1.000000"
    Sentera:AlignXYDeg: "0.010"
    Sentera:Session: "2020-07-19_17-23-11"
    Sentera:VignettingCenter: "768.00"
    Sentera:VignettingEnabled: "1"
    Sentera:VignettingPolynomial: "6.5e-10"
    tiff:Compression: 1
    tiff:PhotometricInterpretation: 1
    tiff:PlanarConfiguration: 1
    tiff:RowsPerStrip: 2
 subimage  3: 1904 x 1428, 1 channel, uint12 tiff
    channel list: Y
    compression: "none"
    DateTime: "2020:07:19 17:36:28"
    ExposureTime: 0.001308
    FNumber: 1.8
    Make: "Sentera"
    Model: "21214-01_3.2MP-GS-0001"
    Orientation: 1 (normal)
    PixelAspectRatio: 1
    planarconfig: "contig"
    ResolutionUnit: "in"
    Software: "1.3.2"
    XResolution: 238
    YResolution: 238
    Camera:AboveGroundAltitude: "111.612001"
    Camera:BandName: "RedEdge"
    Camera:BandSensitivity: "0.370"
    Camera:CentralWavelength: "715"
    Camera:FlightUniqueID: "2020-07-19_17-23-11"
    Camera:GPSXYAccuracy: "0.660000"
    Camera:GPSZAccuracy: "1.017000"
    Camera:ImageUniqueID: "5010"
    Camera:IsNormalized: "0"
    Camera:ModelType: "perspective"
    Camera:PerspectiveDistortion: "-0.117,0.179,-0.045,0.000,0.000"
    Camera:PerspectiveFocalLength: "7.800"
    Camera:PerspectiveFocalLengthUnits: "mm"
    Camera:Pitch: "0.000000"
    Camera:PrincipalPoint: "3.305,2.460"
    Camera:RigCameraIndex: "3"
    Camera:RigName: "Sentera-6x"
    Camera:RigRelatives: "0.000,0.000,0.000"
    Camera:Roll: "0.000000"
    Camera:SunSensor: "12760.552"
    Camera:WavelengthFWHM: "10"
    Camera:Yaw: "90.000000"
    Exif:ApertureValue: 1.69599 (f/1.8)
    Exif:BodySerialNumber: "047"
    Exif:DateTimeDigitized: "2020:07:19 17:36:28"
    Exif:DateTimeOriginal: "2020:07:19 17:36:28"
    Exif:ExposureMode: 0 (auto)
    Exif:FocalLength: 8 (8 mm)
    Exif:FocalPlaneResolutionUnit: 3 (cm)
    Exif:FocalPlaneXResolution: 2898.55
    Exif:FocalPlaneYResolution: 2898.55
    Exif:LensModel: "8.00mm-0005_0023"
    Exif:MeteringMode: 1 (average)
    Exif:SceneCaptureType: 0 (standard)
    Exif:SubsecTime: "125488"
    Exif:SubsecTimeDigitized: "125488"
    Exif:SubsecTimeOriginal: "125488"
    oiio:BitsPerSample: 12
    oiio:ColorSpace: "sRGB"
    Sentera:AlignEnabled: "1"
    Sentera:AlignMatrix: "1.000000"
    Sentera:AlignXYDeg: "-0.234"
    Sentera:Session: "2020-07-19_17-23-11"
    Sentera:VignettingCenter: "768.00"
    Sentera:VignettingEnabled: "1"
    Sentera:VignettingPolynomial: "6.5e-10"
    tiff:Compression: 1
    tiff:PhotometricInterpretation: 1
    tiff:PlanarConfiguration: 1
    tiff:RowsPerStrip: 2
 subimage  4: 1904 x 1428, 1 channel, uint12 tiff
    channel list: Y
    compression: "none"
    DateTime: "2020:07:19 17:36:28"
    ExposureTime: 0.001205
    FNumber: 1.8
    Make: "Sentera"
    Model: "21214-01_3.2MP-GS-0001"
    Orientation: 1 (normal)
    PixelAspectRatio: 1
    planarconfig: "contig"
    ResolutionUnit: "in"
    Software: "1.3.2"
    XResolution: 238
    YResolution: 238
    Camera:AboveGroundAltitude: "111.612001"
    Camera:BandName: "NIR"
    Camera:BandSensitivity: "0.246"
    Camera:CentralWavelength: "840"
    Camera:FlightUniqueID: "2020-07-19_17-23-11"
    Camera:GPSXYAccuracy: "0.660000"
    Camera:GPSZAccuracy: "1.017000"
    Camera:ImageUniqueID: "5010"
    Camera:IsNormalized: "0"
    Camera:ModelType: "perspective"
    Camera:PerspectiveDistortion: "-0.117,0.179,-0.045,0.000,0.000"
    Camera:PerspectiveFocalLength: "7.800"
    Camera:PerspectiveFocalLengthUnits: "mm"
    Camera:Pitch: "0.000000"
    Camera:PrincipalPoint: "3.231,2.481"
    Camera:RigCameraIndex: "4"
    Camera:RigName: "Sentera-6x"
    Camera:RigRelatives: "0.000,0.000,0.000"
    Camera:Roll: "0.000000"
    Camera:SunSensor: "7326.333"
    Camera:WavelengthFWHM: "20"
    Camera:Yaw: "90.000000"
    Exif:ApertureValue: 1.69599 (f/1.8)
    Exif:BodySerialNumber: "047"
    Exif:DateTimeDigitized: "2020:07:19 17:36:28"
    Exif:DateTimeOriginal: "2020:07:19 17:36:28"
    Exif:ExposureMode: 0 (auto)
    Exif:FocalLength: 8 (8 mm)
    Exif:FocalPlaneResolutionUnit: 3 (cm)
    Exif:FocalPlaneXResolution: 2898.55
    Exif:FocalPlaneYResolution: 2898.55
    Exif:LensModel: "8.00mm-0005_0024"
    Exif:MeteringMode: 1 (average)
    Exif:SceneCaptureType: 0 (standard)
    Exif:SubsecTime: "125538"
    Exif:SubsecTimeDigitized: "125538"
    Exif:SubsecTimeOriginal: "125538"
    oiio:BitsPerSample: 12
    oiio:ColorSpace: "sRGB"
    Sentera:AlignEnabled: "1"
    Sentera:AlignMatrix: "1.000000"
    Sentera:AlignXYDeg: "0.010"
    Sentera:Session: "2020-07-19_17-23-11"
    Sentera:VignettingCenter: "768.00"
    Sentera:VignettingEnabled: "1"
    Sentera:VignettingPolynomial: "6.5e-10"
    tiff:Compression: 1
    tiff:PhotometricInterpretation: 1
    tiff:PlanarConfiguration: 1
    tiff:RowsPerStrip: 2

```

Note: I cannot publicly post the TIFF file.

**Platform information:**
 - OIIO branch/version: 2.2.11
 - OS: Pop!_OS 20.04 LTS
 - C++ compiler: 9.3.0
 - LIBTIFF Version 4.1.0
 - Any non-default build flags when you build OIIO:

**IF YOU ALREADY HAVE A CODE FIX:** There is no need to file a separate issue,
please just go straight to making a [pull request](https://github.com/OpenImageIO/oiio/pulls).

**Comment by lgritz** [comment]
I'm not sure how to address this without the example file.

Can you send it to me personally (just not post publicly)?

Is it just the image content you can't post? Can you create another image that contains the same metadata, but all black pixels, and post that?

**Comment by chippey** [comment]
Sending you the file privately would work.  What email address should I send it to?

**Comment by lgritz** [comment]
lg@larrygritz.com is fine

**Comment by chippey** [comment]
Sent it, hopefully it went through okay as the file is a little big for email (~24MB)

Cheers

**Comment by lgritz** [comment]
I see what's happening. When parsing XMP data in the header, it's not turning sequences/lists into arrays, it's just recording one value. Working on a fix.

**Comment by lgritz** [comment]
Fix coming. Now reports as:

    Sentera:AlignEnabled: 1
    Sentera:AlignMatrix: 0.999999, 0.001538, 77.1999, -0.001538, 0.999999, 53.255, 0, 0, 1
    Sentera:AlignXYDeg: 78.3, 51.8, -0.088
    Sentera:Session: "2020-07-19_17-23-11"
    Sentera:VignettingCenter: 1024, 768
    Sentera:VignettingEnabled: 1
    Sentera:VignettingPolynomial: 0, 0, 6.5e-10

#2865

**Comment by chippey** [comment]
Huzzah!  Thanks!

Ahh, the joys of parsing XML...
Although this comes to mind as well in terms of EXIF,XMP, etc... https://xkcd.com/927/

**Comment by lgritz** [comment]
Fixed by #2865
